pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,182 @@
+              type: string
+            userId:
+              type: string
+              EOT`}
+</CodeBlock>
+
+2. Validate incoming MQTT messages by loading the AsyncAPI schema definition via the `fromSource` method. 
+
+<CodeBlock>
+{`const AsyncApiValidator = require('asyncapi-validator')
+let va = await AsyncApiValidator.fromSource('./api.yaml')
+// validate messageId 'UserRemoved'
+va.validateByMessageId('UserRemoved', {
+  userId: '123456789',
+  userEmail: 'alex@mail.com',
+})`}
+</CodeBlock>
+
+## Validation in a gateway
+A gateway intercepts all incoming messages and routes them through the middleware and handler pipelines. The gateway sits between the producer and the broker. The messages are routed through the gateway first, and the gateway determines whether the message is valid. If the message is invalid, it displays an error and is not forwarded to the broker.
+
+<Remember>
+Currently, only the Kafka protocol is supported.
+</Remember>
+
+```mermaid
+graph TD
+    PR[Producer]-- Message --- EG[AsyncAPI Event-Gateway]
+    EG -- Message --- EGV{Is Message valid?}
+    EGV -->|Yes| BR[Broker]
+    EGV --- |No| INV{Fail when invalid?}
+    INV -->|Yes| ERR[/Fail/] -- Produce request errored --> PR
+    INV -->|No| BR
+```
+The AsyncAPI document is important in this case because payload schemas are taken from it and messages as validated against it in your application.
+You can spin up the AsyncAPI gateway using an AsyncAPI file. All the messages are forwarded to a WebSocket endpoint; if the message/payload is invalid, it includes a validation error message.
+
+### UseCase
+You can use a Kafka consumer/producer[(kcat)](https://github.com/edenhill/kcat), a broker, and a simple WebSocket to run the AsyncAPI gateway in your machine.
+```
+type: object
+properties:
+  lumens:
+    type: integer
+    minimum: 0
+    description: Light intensity measured in lumens.
+  sentAt:
+    type: s
+    format: date-time
+    description: Date and time when the message was sent.
+ ```
+Here the expected payload `lightMeasured` is an integer. A validation error will be included if the producer sends a message with the type boolean.
+ ```
+ "_asyncapi_eg_validation_error": "{\"ts\":\"2021-12-20T11:33:26.583143572Z\",\"errors\":[\"lumens: Invalid type. Expected: integer, given: boolean\"]}",
+ ```
+## Validation handled by the broker
+As producers and consumers do not communicate with each other directly, but rather information transfer happens via Kafka topic. At the same time, the consumer still needs to know the type of data the producer is sending. Imagine if the producer starts sending bad data to Kafka or if the data type of your data gets changed. We need a way to have a common data type that must be agreed upon.
+
+That’s where Schema Registry comes into the picture. It is an application that resides outside of your Kafka cluster and handles the distribution of schemas to the producer and consumer by storing a copy of schema in its local cache.
```

Comment:
```suggestion
That’s where Schema Registry comes into the picture. It is an application that resides outside of your Kafka cluster and handles the distribution of schemas to the producer and consumer by storing a copy of schema in its local cache.
```

`It is an application that resides outside of your Kafka cluster` you are sure about it? that it is outside the cluster? I'm not tbh

Reply:
I'm not sure but I searched about it, I found it in a blog written by someone about schema registry

Reply:
can you give us a link here in this PR to the blog post you're referring you read about this? would be good to see the direct quote and confirm 😄✌🏽 @starlightknown

Reply:
https://docs.confluent.io/platform/current/schema-registry/index.html#schemas-subjects-and-topics

Reply:
![image](https://user-images.githubusercontent.com/74637789/206887417-1bb322c9-ac42-41d8-9ceb-bf9fed18a37f.png)
here @alequetzalli @derberg